pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -218,6 +224,19 @@ public void init(SegmentGeneratorConfig config, SegmentCreationDataSource dataSo
     LOGGER.debug("tempIndexDir:{}", _tempIndexDir);
   }
 
+  // input int[] can be used to map sortedDocIds[immutableId] = mutableId (based on RecordReader iteration order)
+  // output int[] should be used to map output[mutableId] = immutableId
+  private int[] convertSortedDocIds(@Nullable int[] sortedDocIds) {
```

Comment:
sortedDocIds makes people think the docIds in the array is sorted. In this case it is not -- you probably mean immutableId is sorted as array index. Can we find a better name here with some explanation and example? e.g., immutableIdToMutableId map.

Reply:
IIUC, if the table have sort index, then the docIds need to be sorted accordingly when generate immutable segment?